JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/CompileAndRunBean.java

```diff
@@ -1,27 +1,200 @@
+import ru.ncedu.dynamic.DynamicCompiler;
+import ru.ncedu.dynamic.StringJavaFileObject;
+
 import javax.ejb.Stateless;
 import javax.inject.Named;
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
+import javax.tools.SimpleJavaFileObject;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Created by artyom on 15.12.15.
  */
 @Stateless
 @Named(value = "compileAndRunBean")
-public class CompileAndRunBean {
+public class CompileAndRunBean extends DynamicCompiler {
 
     @PersistenceContext
     EntityManager em;
 
+
+    public static final String DEFAULT_PACKAGE_PREFFIX = "";
+    private static final Logger LOG = Logger.getLogger(CompileAndRunBean.class);
+
+    public void init(ClassLoader parentLoader) {
+        try {
+            super.init(parentLoader);
+        } catch (Exception e) {
+            LOG.error("Exception", e);
+        }
+    }
+
     //!TODO
-    public String compile(String projectHash){
-        return null;
+    public String compile(String projectHash) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream(baos);
+        PrintStream old = System.out;
+        System.setOut(printStream);
+        BasicConfigurator.configure();
+        List<File> sources;
+        FileBean fileBean = new FileBean();
+        sources = fileBean.getProjectFiles(projectHash);
+        try {
+            init(ClassLoader.getSystemClassLoader());
+        } catch (NullPointerException e) {
+            LOG.error("NullPointerException in getSystemClassLoader()", e);
+        }
+        List<SimpleJavaFileObject> userSources = new ArrayList<>();
+        for (int i = 0; i < sources.size(); i++) {
+            File file = sources.get(i);
+            // typeId = 8 - "class", typeId = 9 - "interface", typeId = 10 - "exception"
+            // typeId = 5 - "runnable", typeId = 11 - "enum", typeId = 12 - "annotation"
+            int typeId = file.getType().getTypeId();
+            if (typeId == 8 || typeId == 9 || typeId == 10 || typeId == 5 || typeId == 11 || typeId == 12) {
+                byte[] userFByte = file.getData();
+                String userFile = "";
+                for (int j = 0; j < userFByte.length; j++) {
+                    userFile += (char) userFByte[j];
+                }
+                String s = findSrc(sources);
+                String src = file.getPath();
+                src = src.replaceFirst(s + "/", "");
+                src = src.replace("/", ".");
+                int pos = src.lastIndexOf(file.getFileName());
+                String pack = src.substring(0, pos);
+                String className = file.getFileName() + ".java";
+                SimpleJavaFileObject sjf = null;
+                if (pack != null) {
```

Comment:
This block can be put into constructResource(String, String, String). So you won't have constructResource(String, String) anymore.